lib/scout_apm/tracked_request.ex

```diff
@@ -63,6 +63,7 @@ defmodule ScoutApm.TrackedRequest do
     # We finished tracing this request, so go and record it.
     if Enum.count(layers(tr2)) == 0 do
       request = tr2 |> with_root_layer(updated_layer)
+      ScoutApm.DevTrace.Store.record(request)
```

Comment:
I'd do this in collector - ideally `TrackedRequest` doesn't know much about what happens to it after it gets handed off. (plus makes testing this fiddly data structure easier).